### DISCUSSION TITLE
Axis direction.

### DISCUSSION THREAD

**Discussion by vegetate7** [kachurovskiy/nanoels#150] [General]
Hello! I just realized what I count Z coordinate from left to right (from chuck to tailstock) all years. And nanoEls do it oppozite.
I did some search, and no strict convention about it. Even CNC have a left hand and right hand directions. 
So what is the real final true directions of axis? 
Is it hard to add reverse logic to firmware? (I guess yes...)

**Comment by kachurovskiy** [comment]
Hi @vegetate7, there's a relation between pitch and axis position - positive pitch and positive angle change produce positive change in axis position. GCode instructions also set axis position assuming positive is towards the chuck. Changing axis direction will damage all that logic and I can't think of a good way to make it adjustable.

In lathecode we now support any axis direction for the generated GCode via AXES directive - https://github.com/kachurovskiy/lathecode?tab=readme-ov-file#axis-direction

**Comment by duebelpapst** [comment]
Hello,

I'm currently playing around with lathegcode, and I'm having a hard time with the fact that the axis directions are reversed from what I'm used to with industrial machines.
But it works.
However, as I tend to machine long parts, the removal by grooving is very slow, so I created the Gcode with Fusion, where I can also set longitudinal turning.
Unfortunately, before I load the code into the controller, I have to invert all X and Y values by hand, which is inconvenient.
Does anyone know a way to get Fusion to invert the axis directions?
I have tried all possible post processors without success. 
Otherwise I'll just have to program something.

**Reply by RiickDEV** [reply]
Hello, I had the same problem. When running a file, the tool moves towards the material in the positive Z direction, and also advances into the material in the positive X direction.

I solved this problem by inverting the values ​​in the post-processing file. I'm using GRBL Turning.

Here is the file that automatically generates the inverted X and Z values:
[grbl_turning_invert.txt](https://github.com/user-attachments/files/23148033/grbl_turning_invert.txt)

rename to .cps and move to posts folder

C:\Users\x\AppData\Roaming\Autodesk\Fusion 360 CAM\Posts

**Reply by kachurovskiy** [reply]
In lathecode we now support any axis direction for the generated GCode via AXES directive - https://github.com/kachurovskiy/lathecode?tab=readme-ov-file#axis-direction
